### ISSUE TITLE
[PoX-4] Match function signature for `extend` methods

### ISSUE THREAD

**Issue by janniks** [stacks-network/stacks-core#4230]
Currently _(`next` branch at time of writing),_ the methods: `stack-extend` and `delegate-stack-extend` have slightly differing signatures, even though they overlap in arguments a lot.
Additionally, all other `signer-key` accepting stacking methods have the `signer-key` as their last argument.

https://github.com/stacks-network/stacks-core/blob/3e21883fba3e80a98a849b7f159df0db913b662b/stackslib/src/chainstate/stacks/boot/pox-4.clar#L963-L965

https://github.com/stacks-network/stacks-core/blob/3e21883fba3e80a98a849b7f159df0db913b662b/stackslib/src/chainstate/stacks/boot/pox-4.clar#L1138-L1142

Might this be a good time to change the `delegate-stack-extend` signature (argument ordering)? Since, with the pox-4 update users will have to revisit any breaking code already.

to e.g.
```clar
(define-public (delegate-stack-extend
                    (stacker principal)
                    (extend-count uint)
                    (pox-addr { version: (buff 1), hashbytes: (buff 32) })
                    (signer-key (buff 33)))
```

CC @friedger 

> Haven't found anything exact on this in the SIPs, but I might be missing something.

**Comment by friedger** [comment]
The proposed change makes sense to me. Let's do it. @setzeus

**Comment by setzeus** [comment]
ACK. I don't see any reason we can't do this & indeed remember coming across the different order, thanks for the note @janniks.

**Comment by jcnelson** [comment]
Hi, we don't know who out there would be affected by a change to the public API signature, so we can't assess what the fallout would be of changing the arguments.  If you feel really strongly about the aesthetics, then please add a separate public function that just calls this one (but don't change the existing one).

**Comment by janniks** [comment]
This would only be changing the signatures, which are already being changed from pox-3 → pox-4 (i.e. adding the signer-key argument)

**Comment by jcnelson** [comment]
It is a lot easier to change the contract than the signatures, especially
if the former is already expected but the latter isn't

**Comment by friedger** [comment]
Developers/users need to change the signature already when switching to pox-4. The api is not yet released in one way or the other.

**Comment by janniks** [comment]
Partially closed by #4279 -- I'm fine to close, at least `signer-key` will be consistently last. The other params I'm okay with keeping the same as `pox-3`
